### PR TITLE
Finishing up rvars for CRAN

### PR DESCRIPTION
**Pull request by mjskay**
This PR finishes up the last bit of changes from #117.

## Support multidimensional arrays with `cdf.rvar()`, `quantile.rvar()`, and `density.rvar()`

These follow the same pattern as `range.rvar()`: they place their output in the first dimension. Examples:

``` r
set.seed(1234)
x = c(rvar(rnorm(100, mean = 0)), rvar(rnorm(100, mean = 1)))
x
#> rvar<100>[2] mean ± sd:
#> [1] -0.16 ± 1   1.04 ± 1
density(x, seq(-2, 2, length.out = 5))
#>            [,1]       [,2]
#> [1,] 0.05268921 0.00000000
#> [2,] 0.36854780 0.04935177
#> [3,] 0.33059940 0.22144520
#> [4,] 0.16166408 0.42742687
#> [5,] 0.07011558 0.19442184
cdf(x, seq(-2, 2, length.out = 5))
#>      [,1] [,2]
#> [1,] 0.02 0.00
#> [2,] 0.20 0.03
#> [3,] 0.65 0.12
#> [4,] 0.86 0.49
#> [5,] 0.96 0.85
quantile(x, ppoints(5))
#>            [,1]       [,2]
#> [1,] -1.1599108 0.01535116
#> [2,] -0.7144907 0.61815900
#> [3,] -0.3846280 1.03280328
#> [4,]  0.2562909 1.44069893
#> [5,]  1.0882675 2.36920578
```

## `rvar_quantile()` for getting quantiles as random variables

One difference from `stats::quantile()` is that I have the default of `names` be `FALSE` instead of `TRUE` as I think the default being `TRUE` causes more problems than it is worth, though I am not wedded to this choice. Example:

``` r
set.seed(5678)
x = rvar_rng(rnorm, 4, mean = 1:4, sd = 2)
x
#> rvar<4000>[4] mean ± sd:
#> [1] 0.98 ± 2  1.98 ± 2  2.96 ± 2  4.02 ± 2
rvar_median(x)
#> rvar<4000>[1] mean ± sd:
#> [1] 2.5 ± 1.1
rvar_quantile(x, probs = c(0.25, 0.5, 0.75))
#> rvar<4000>[3] mean ± sd:
#> [1] 1.3 ± 1.2  2.5 ± 1.1  3.6 ± 1.2
rvar_quantile(x, probs = c(0.25, 0.5, 0.75), names = TRUE)
#> rvar<4000>[3] mean ± sd:
#>        25%        50%        75% 
#> 1.3 ± 1.2  2.5 ± 1.1  3.6 ± 1.2
```

## `str.rvar()` now shows attributes

I have updated `str.rvar()` to show attributes of the rvar object and the underlying draws array (unless the meaning of these attributes are already included in the output, e.g. number of chains). I still keep the output hiding the internal structure (e.g. it doesn't reveal that draws are stored as an attribute on the object, instead showing usage of `draws_of()` to retrieve it). Example:

``` r
x <- rvar(array(1:24, dim = c(2,3,4)))
str(x)
#>  rvar<2>[3,4]  1.5 ± 0.71  3.5 ± 0.71  5.5 ± 0.71  7.5 ± 0.71 ...
x_with_attrs <- x
dimnames(x_with_attrs)[1] = list(c("a","b","c"))
attr(draws_of(x_with_attrs), "foo") = list(1,2)
attr(x_with_attrs, "bar") = list(1,2)
str(x_with_attrs)
#>  rvar<2>[3,4]  1.5 ± 0.71  3.5 ± 0.71  5.5 ± 0.71  7.5 ± 0.71 ...
#>  - dimnames(*)=List of 2
#>   ..$ : chr [1:3] "a" "b" "c"
#>   ..$ : NULL
#>  - attr(draws_of(*), "foo")=List of 2
#>   ..$ : num 1
#>   ..$ : num 2
#>  - attr(*, "bar")=List of 2
#>   ..$ : num 1
#>   ..$ : num 2
```

I think that this plus #153 will finish up everything from #117 giving us a solid version of rvar for the first CRAN release.

### CODE REVIEW

File: R/rvar-dist.R [hunk 1]
```diff
@@ -14,37 +14,55 @@
...
 #' @return
 #'
-#' A vector of the same length as the input (`q`, `at`, or `probs`) containing
-#' values from the corresponding function of the given [`rvar`].
+#' If `x` is a scalar [`rvar`], returns a vector of the same length as the input
+#' (`q`, `at`, or `probs`) containing values from the corresponding function
+#' of the given [`rvar`].
 #'
-#' @name rvar-functions
+#' If `x` has length greater than 2, returns an array with dimensions
```
Comment (paul-buerkner) [inline R/rvar-dist.R:21]: do you mean larger than 1?

File: R/rvar-print.R [hunk 2]
```diff
@@ -65,11 +72,25 @@ format.rvar <- function(x, ..., summary = NULL, digits = 2, color = FALSE) {
...
 #' @rdname print.rvar
 #' @export
-str.rvar <- function(object, ..., summary = NULL, vec.len = NULL) {
+str.rvar <- function(
+  object, ..., summary = NULL, vec.len = NULL,
+  indent.str = paste(rep.int(" ", max(0, nest.lev + 1)), collapse = ".."),
+  nest.lev = 0, give.attr = TRUE
+) {
+
+  str_next <- function(x, ...) str(x, ...,
```
Comment (paul-buerkner) [inline R/rvar-print.R:81]: the indentation is a little bit difficult to read. Is it worth adding formal `{}` to the function definition?

**Review by paul-buerkner — COMMENTED** [review]
Looks good to me! I just had two minor comments. Feel free to address them as you please and merge away :-)

File: R/rvar-dist.R (hunk 1 above)
Comment (mjskay) [reply R/rvar-dist.R:21]: hahaha yuuup :)

File: R/rvar-print.R (hunk 2 above)
Comment (mjskay) [reply R/rvar-print.R]: sure, will do

**Comment by mjskay** [comment]
Thanks so much! I addressed the two minor issues you raised, so once the checks pass I will merge.

**Pull request merged**